backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/appsearchmodal/internal/BpkAppSearchModalImpl.kt

```diff
@@ -0,0 +1,90 @@
+/**
+ * Backpack for Android - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.compose.appsearchmodal.internal
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import net.skyscanner.backpack.compose.appsearchmodal.AppSearchModalResult
+import net.skyscanner.backpack.compose.icon.BpkIcon
+import net.skyscanner.backpack.compose.sleketon.BpkBodyTextSkeleton
+import net.skyscanner.backpack.compose.textfield.BpkTextField
+import net.skyscanner.backpack.compose.tokens.BpkSpacing
+import net.skyscanner.backpack.compose.tokens.Search
+
+private const val SkeletonCount = 10
+private const val SkeletonItemWidth = 200
+
+@Composable
+internal fun BpkAppSearchModalImpl(
+    inputText: String,
+    inputHint: String,
+    results: AppSearchModalResult,
+    onInputChanged: (String) -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    when (results) {
+        is AppSearchModalResult.Error -> {
+            BpkSearchModalError(results = results)
+        }
+
+        else -> {
+            Column(modifier = modifier) {
+                BpkTextField(
+                    icon = BpkIcon.Search,
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .padding(BpkSpacing.Base),
+                    value = inputText,
+                    placeholder = inputHint,
+                    onValueChange = onInputChanged,
+                )
+                LazyColumn {
+                    if (results is AppSearchModalResult.Content) {
+                        results.shortcuts?.let {
+                            item {
+                                BpkShortcuts(it)
+                            }
+                        }
+                        items(results.sections) {
```

Comment:
from a lazy perspective it might be better to unbox the sections here and have the section header as a separate item and then each item as an item otherwise we don't really gain a lot from this being lazy depending on the section sizes/count